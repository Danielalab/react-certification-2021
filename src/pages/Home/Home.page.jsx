import React from 'react';
import styled from 'styled-components';
import VideoThumbnail from '../../components/VideoThumbnail';
import useSearchQuery from '../../hooks/useSearchQuery';

const Wrapper = styled.section`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 2rem;
  margin-top: 70px;
  @media (max-width: 576px) {
    justify-content: center;
    padding: 1rem;
  }
`;

function HomePage({ text }) {
  const [data, loading, error] = useSearchQuery({ text });
  return (
    <Wrapper>
      {loading && '...loading'}
      {error && 'Something was wrong'}
      {data &&
        data.items
          .filter(({ snippet }) => snippet !== undefined)
          .map(({ id, snippet }) => (
            <VideoThumbnail key={id.videoId} id={id.videoId} videoData={snippet} />
          ))}
    </Wrapper>
  );
}

export default HomePage;
