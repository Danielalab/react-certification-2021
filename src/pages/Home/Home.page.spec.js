import Home from './Home.page';
import dataDummy from '../../hooks/tests/responses/searchQuerySuccess.json';

const { render, screen, waitFor } = require('@testing-library/react');

jest.mock('../../components/VideoThumbnail/VideoThumbnail.component.jsx');

describe('Home Page', () => {
  test('Should renders Home page component with VideoThumbnail', async () => {
    let loading;
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(dataDummy),
      })
    );
    const { container } = render(<Home text="fake searched text" />);
    loading = screen.getByText(/...loading/i);
    expect(container).toBeInTheDocument();
    expect(loading).toBeInTheDocument();
    expect(container).toMatchSnapshot();

    await waitFor(() => screen.getAllByText(/dummy video thumbnail component/i));
    const videosThumbnailComponent = screen.getByText(
      /dummy video thumbnail component fakeVideoId1/i
    );
    expect(videosThumbnailComponent).toBeInTheDocument();
    expect(videosThumbnailComponent.textContent).toBe(
      'Dummy Video Thumbnail component fakeVideoId1'
    );
    loading = screen.queryByText(/...loading/i);
    expect(loading).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should renders an Error message', async () => {
    let loading;
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.reject(new Error('API error')),
      })
    );
    const { container } = render(<Home text="fake searched text" />);
    loading = screen.getByText(/...loading/i);
    expect(container).toBeInTheDocument();
    expect(loading).toBeInTheDocument();
    expect(container).toMatchSnapshot();

    await waitFor(() => screen.getByText(/Something was wrong/i));
    const errorMessage = screen.getByText(/Something was wrong/i);
    expect(errorMessage).toBeInTheDocument();
    loading = screen.queryByText(/...loading/i);
    expect(loading).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
