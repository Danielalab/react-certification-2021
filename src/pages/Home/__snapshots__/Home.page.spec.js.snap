// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Should renders Home page component with VideoThumbnail 1`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    ...loading
  </section>
</div>
`;

exports[`Home Page Should renders Home page component with VideoThumbnail 2`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    <div
      data-testid="video-thumbnail"
    >
      <p>
        Dummy Video Thumbnail component 
        fakeVideoId1
      </p>
    </div>
    <div
      data-testid="video-thumbnail"
    >
      <p>
        Dummy Video Thumbnail component 
        fakeVideoId2
      </p>
    </div>
  </section>
</div>
`;

exports[`Home Page Should renders an Error message 1`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    ...loading
  </section>
</div>
`;

exports[`Home Page Should renders an Error message 2`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    Something was wrong
  </section>
</div>
`;

exports[`Home Page Should renders an Error message 1`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    ...loading
  </section>
</div>
`;

exports[`Home Page Should renders an Error message 2`] = `
<div>
  <section
    class="sc-bdnxRM hyfyNv"
  >
    Something was wrong
  </section>
</div>
`;
